pages/api/internal/assets/[anchor]/[assetType]/index.js

```diff
@@ -114,22 +115,27 @@ export default async function handle(req, res) {
 		}
 	})
 
-	const contract = nft.contract
 
 	// TODO: What to do with assets that aren't attached to any other NFT?
 	// Should this clean up run on an async job?
 	await dropAssetsFromNFT(nft)
 
 	const form = new formidable.IncomingForm()
+	// This is a temporary directory for uploads
+	// It allows us to cache files, but at the same time is isolated from the system's /tmp
+	// this should prevent Arbitrary Code Execution attacks via Image-Uploads.
+	const tmpDir = path.join(process.env.STORAGE_DIR, "upload") 
+	form.uploadDir = tmpDir
+
 	form.parse(req, async (err, fields, files) => {
 		const uploadedFile = files.assets
-
+		
 		if (!allowedMimeTypes.includes(uploadedFile.type)) {
-			return res.status(422).json({ message: 'File type is not allowed' })
+			rmSync(uploadedFile.path)
+			return res.status(422).json({ message: 'File type ${uploadedFile.type} is not allowed' })
 		}
 
-		const filePath = uploadedFile.path
-		const fileContent = readFileSync(filePath)
+		const fileContent = readFileSync(uploadedFile.path)
```

Comment:
You're reading the entire file into RAM here. The application's memory usage will increase accordingly during uploads. To avoid the container being killed for large uploads you must stream your file to the hashing function instead.

Reply:
Noted for later: https://github.com/authenticvision/digitalsoul/issues/42

Reply:
Fine with me, nitpicking tho: Your description in #42 is "Stream files during upload instead of reading them to memory" but it's after the upload and it should probably mention that this is to fix excessive memory usage during file hashing.